benchmarks/benchmarkInnerProductKernels.cpp

```diff
@@ -24,77 +24,109 @@ namespace LvArray
 namespace benchmarking
 {
 
-#define INNER_PRODUCT_KERNEL( a_i, b_i ) \
+#define INNER_PRODUCT_KERNEL( N, a_i, b_i ) \
   VALUE_TYPE sum = 0; \
   for( INDEX_TYPE i = 0; i < N; ++i ) \
   { \
     sum += a_i * b_i; \
   } \
   return sum
 
-#define INNER_PRODUCT_KERNEL_RAJA( a_i, b_i ) \
+#define INNER_PRODUCT_KERNEL_RAJA( N, a_i, b_i ) \
   RAJA::ReduceSum< typename RAJAHelper< POLICY >::ReducePolicy, VALUE_TYPE > sum( 0 ); \
-  RAJA::forall< POLICY >( RAJA::TypedRangeSegment< INDEX_TYPE >( 0, N ), \
-                          [sum, a, b] LVARRAY_HOST_DEVICE ( INDEX_TYPE const i ) { \
+  forall< POLICY >( N, [sum, a, b] LVARRAY_HOST_DEVICE ( INDEX_TYPE const i ) \
+  { \
     sum += a_i * b_i; \
```

Comment:
```suggestion
    sum = sum + a_i * b_i; \
```

Reply:
The `RAJA::ReduceSum` type doesn't have an `operator=` so this will have to suffice.